assets/js/googlesitekit/datastore/modules/modules.js

```diff
@@ -0,0 +1,434 @@
+					...state.isFetchingSetModuleStatus,
+					[ slug ]: true,
+				},
+			};
+		}
+
+		case FINISH_FETCH_SET_MODULE_STATUS: {
+			const { slug } = payload;
+
+			return {
+				...state,
+				isFetchingSetModuleStatus: {
+					...state.isFetchingSetModuleStatus,
+					[ slug ]: false,
+				},
+			};
+		}
+
+		case CATCH_FETCH_SET_MODULE_STATUS: {
+			const { slug } = payload;
+
+			return {
+				...state,
+				error: payload.error,
+				isFetchingSetModuleStatus: {
+					...state.isFetchingSetModuleStatus,
+					[ slug ]: false,
+				},
+			};
+		}
+
+		case START_FETCH_MODULES: {
+			return {
+				...state,
+				isFetchingModules: true,
+			};
+		}
+
+		case RECEIVE_MODULES: {
+			const { modules } = payload;
+
+			return {
+				...state,
+				modules: modules.reduce( ( acc, module ) => {
+					return { ...acc, [ module.slug ]: module };
+				}, {} ),
+			};
+		}
+
+		case FINISH_FETCH_MODULES: {
+			return {
+				...state,
+				isFetchingModules: false,
+			};
+		}
+
+		case CATCH_FETCH_MODULES: {
+			return {
+				...state,
+				error: payload.error,
+				isFetchingModules: false,
+			};
+		}
+
+		default: {
+			return { ...state };
+		}
+	}
+};
+
+export const resolvers = {
+	*getModules() {
+		const registry = yield Data.commonActions.getRegistry();
+
+		const existingModules = registry.select( STORE_NAME ).getModules();
+
+		if ( ! existingModules ) {
+			yield actions.fetchModules();
+		}
+	},
+};
+
+export const selectors = {
+	/**
+	 * Get the list of modules registered for use with Site Kit.
+	 *
+	 * A module is a section of Site Kit that relates to a particular service,
+	 * like Google Analytics or Google PageSpeed modules. They can provide
+	 * admin-only features (like PageSpeed Insights), frontend-only features,
+	 * or both (eg. Analytics, which can install Analytics <script> tags in the
+	 * frontend, and show dashboards in the WordPress Admin).
+	 *
+	 * Returns a hash of objects with the shape, keyed by slug, when successful:
+	 * ```
+	 * slug: {
+	 *   "slug": "tagmanager",
+	 *   "name": "Tag Manager",
+	 *   "description": "Tag Manager creates an easy to manage way to create tags on your site without updating code.",
+	 *   "homepage": "https://tagmanager.google.com/",
+	 *   "internal": false,
+	 *   "active": false,
+	 *   "connected": false,
+	 *   "dependencies": [
+	 *     "analytics"
+	 *   ],
+	 *   "dependants": []
+	 * }
+	 * ```
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @return {Object|undefined} Modules available on the site.
+	 */
+	getModules( state ) {
+		const { modules } = state;
+
+		return modules;
+	},
+
+	/**
+	 * Get a specific module by slug.
+	 *
+	 * Returns a specific module by its slug.
+	 * Returns `undefined` if state is still loading or if said module doesn't exist.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @param {string} slug  Module slug.
+	 * @return {Object|undefined} A specific module object.
```

Comment:
See above, should probably be `?Object`. Also I think it would be good to clarify in the description "..., or null if invalid slug provided."